lld/MachO/SyntheticSections.cpp

```diff
@@ -1204,6 +1205,17 @@ void SymtabSection::emitEndFunStab(Defined *defined) {
   stabs.emplace_back(std::move(stab));
 }
 
+// Given a pointer to a function symbol, return the symbol that points to the
+// actual function body that will go in the final binary. Generally this is the
+// symbol itself, but if the symbol was folded using a thunk, we retrieve the
+// target function body from the thunk.
+Defined *SymtabSection::getFuncBodySym(Defined *originalSym) {
+  if (originalSym->identicalCodeFoldingKind != Symbol::ICFFoldKind::Thunk)
```

Comment:
Oh sorry, I meant that you should flip the condition and the return. So that if it is a thunk, we return the thunk. Otherwise we return the original symbol.